packages/dev-middleware/src/inspector-proxy/Device.js

```diff
@@ -371,6 +421,13 @@ export default class Device {
       // for unknown pages.
       const page: ?Page = this.#pages.get(pageId);
 
+      // NOTE(bycedric): Notify the device message middleware of the disconnect event, without any further actions.
+      // This can be used to clean up state in the device message middleware.
+      this.#getMessageMiddleware(
+        this.#debuggerConnection,
+        page,
+      )?.handleDeviceMessage(message);
```

Comment:
I added a comment just above that on why the return value is not used here:

```
// NOTE(bycedric): Notify the device message middleware of the disconnect event, without any further actions.
// This can be used to clean up state in the device message middleware.
```

But, since we now scope the middleware on the actual device page and debugger connection, I don't think we need to do this anymore. The middleware should be fully garbage collected once either one disconnects. So we don't have to worry about disconnect events.

Reply:
✅ This is now removed, as the whole custom message handler is garbage collected once this happens.